### ISSUE TITLE
List of Stack config files doesn't work

### ISSUE THREAD

**Issue by pauldalewilliams** [amendlik/gitstack-pillar#9]
I'm not sure if I'm missing something, but I can't seem to get this to work.  What I'm trying to do is something similar to the PillarStack example:
```
ext_pillar:  
    - stack:  
      - /path/to/stack1.cfg  
      - /path/to/stack2.cfg
```

So I'm trying this:
```
ext_pillar:
    - git:
      - master git@server.example.com:salt/pillar.git
    - gitstack:
      - master git@server.example.com:salt/pillar.git:
        - stack:
          - _stack/stack_1.cfg
          - _stack/stack_2_roles.cfg
          - _stack/stack_3_minions.cfg
```
But I get an error on the Salt master:
```
2018-09-27 16:07:43,836 [salt.pillar      :972 ][ERROR   ][16772] Execption caught loading ext_pillar 'gitstack':
  File "/usr/lib/python2.7/site-packages/salt/pillar/__init__.py", line 962, in ext_pillar
    key)
  File "/usr/lib/python2.7/site-packages/salt/pillar/__init__.py", line 890, in _external_pillar_data
    *val)
  File "/var/cache/salt/master/extmods/pillar/gitstack.py", line 144, in ext_pillar
    return stack_pillar(minion_id, pillar, *stack_config, **stack_config_kwargs)
  File "/usr/lib/python2.7/site-packages/salt/pillar/stack.py", line 415, in ext_pillar
    if not os.path.isfile(cfg):
  File "/usr/lib64/python2.7/genericpath.py", line 29, in isfile
    st = os.stat(path)

2018-09-27 16:07:44,001 [salt.pillar      :1019][CRITICAL][16772] Pillar render error: Failed to load ext_pillar gitstack: coercing to Unicode: need string or buffer, list found
```

I've tried following the flow but I'm still learning and having trouble figuring out exactly where things are going wrong.  My guess is that it's something in the [section that resolves the paths](https://github.com/amendlik/gitstack-pillar/blob/develop/_pillar/gitstack.py#L115-L134) or in the _resolve_stack function.

I'll try to play around with it later to see if I can figure out what's going wrong but I'd appreciate some help from someone with more expertise.  Thank you!

**Comment by pauldalewilliams** [comment]
I think I've found a way to work around this by using standard pillar for assigning roles to my minions (last entry in stack_1.cfg).  This should still be fixed in order to be on par with PillarStack, but no rush for me at this point.

**Comment by amendlik** [comment]
@pauldalewilliams Thank you for reporting this issue. Please test this fix on the `develop` branch and let me know the results.

**Comment by pauldalewilliams** [comment]
@amendlik Just leaving, but I will test it on Monday.  Thanks for the quick response!

**Comment by pauldalewilliams** [comment]
@amendlik I was able to test this today.  Confirmed:

1.  Still works fine with a single file (e.g., `- stack: _stack/stack.cfg`)
2.  Works with a single file in a list (e.g.:
```
- stack:
  - _stack/stack.cfg
```
3.  Works with a list of multiple files (e.g.:
```
- stack:
  - _stack/stack_1.cfg
  - _stack/stack_2.cfg
```

I didn't test any of the more advanced configs since I don't have any of that (referencing pillars, grains, opts, etc.).

Thanks for this quick fix!  Should be good to close after merging to master.
